### PR TITLE
Add Codecov to CI

### LINKED ISSUE #5: Add Codecov and Coveralls integration to CI
Just like in FerretDB, we should send test coverage information to https://app.codecov.io/gh/FerretDB/wire and https://coveralls.io/github/FerretDB/wire.

### PR DESCRIPTION
**Pull request by noisersup**
Closes #5.
The proof of integration [here](https://app.codecov.io/gh/FerretDB/wire/pulls). As mentioned by the bot, the actual results will be shown after the merge.

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -78,6 +78,23 @@ jobs:
...
+      #
+      # Unfortunately, it seems that tokenless uploads fail too often.
+      # See https://github.com/codecov/codecov-action/issues/837.
+      #
+      # We also can't use ${{ vars.CODECOV_TOKEN }}: https://github.com/orgs/community/discussions/44322
+      - name: Upload coverage information to codecov
+        if: always()
+        uses: codecov/codecov-action@v4
+        with:
+          token: 44625ce5-0586-4c09-93b3-8135f5fc2fb5
```
Comment (AlekSi) [inline .github/workflows/go.yml:92]: Do we still need that token in public?

Comment (noisersup) [reply .github/workflows/go.yml:92]: Well, the discussion related to configuration variables from forks seems to be staled.

The tokenless uploads are failing mostly due to the github rate limiting as far as I'm aware. I guess we need to rely on this [issue](https://github.com/codecov/feedback/issues/301) for now.

Comment (AlekSi) [reply .github/workflows/go.yml:92]: So let's link to that issue instead of a closed one

Comment (noisersup) [reply .github/workflows/go.yml:92]: Done

**Review by AlekSi — APPROVED** [review]

**Review by chilagrow — APPROVED** [review]

**Pull request merged**